### DISCUSSION TITLE
component preview (overview/compositions) is not working after upgrading to bit >=0.0653

### DISCUSSION THREAD

**Discussion by GiladShoham** [teambit/bit#5439] [Help] [answered]
Problem: when opening the overview/compositions tab on bit UI after tagging a component you see a: white screen / failed to load and you get errors on the browser console.

**Comment by GiladShoham** [comment, marked as answer]
**disclaimer**
This solution/issue is relevant to you, only if all the following conditions are met. 
If you have preview problems but these conditions are not met, please open a GitHub issue.
1. You are on a bit version >= 0.0.653
2. You are using custom env
3. Your custom env is based on one of the core envs (`teambit.harmony/node`, `teambit.react/react`, `teambit.react/react-native`, `teambit.html/html`, `teambit.web-components/lit`).
4. You see errors on overview/compositions tabs (in browser console), and they are not rendered.
5. The issue happens for already tagged components (either on bit.dev / self host scope or on the workspace UI when selecting a tagged version on the drop-down in the top right).

---

Lately we enable a new preview architecture (see - https://github.com/teambit/bit/pull/5435)

TL;DR - `preview` step during the build/tag process should be much faster (usually a few seconds instead of about a minute). 
Also, the gallery page should load much faster now.

Full details about the new preview [here](https://github.com/teambit/bit/pull/5247)

Teams using a custom env, which is based on one of these envs - `teambit.harmony/node`, `teambit.react/react`, `teambit.react/react-native`, `teambit.html/html`, `teambit.web-components/lit` will need to rebuild their custom env.

After updating Bit to version 0.0.653 (or later), you need to tag a new version of your environment (even it's not modified), you can do some mock change (like adding a new line). 
This tag process will rebuild your custom env and create all the resources needed.

Once it's ready, you should update your component to use the new env version that you just tagged (this happens automatically if components are on the same workspace as the env), if they are on different workspaces, you will have to update them manually.
(you can use the new `bit aspect update -h` command to do it quickly)

If you have other team members on the team that uses bit, it's really recommended that all of you will upgrade Bit.
